griptape/drivers/vector/opensearch_vector_store_driver.py

```diff
@@ -0,0 +1,189 @@
+from typing import Optional, Union, Tuple
+from opensearchpy import OpenSearch, RequestsHttpConnection
+from griptape import utils
+import logging
+from griptape.drivers import BaseVectorStoreDriver
+from attr import define, field, Factory
+logging.basicConfig(level=logging.ERROR)
+
+
+@define
+class OpenSearchVectorStoreDriver(BaseVectorStoreDriver):
+    host: str = field(kw_only=True)
+    port: int = field(default=443, kw_only=True)
+    http_auth: Optional[Union[str, Tuple[str, str]]] = field(default=None, kw_only=True)
+    use_ssl: bool = field(default=True, kw_only=True)
+    verify_certs: bool = field(default=True, kw_only=True)
+    index_name: str = field(kw_only=True)
+
+    client: OpenSearch = field(default=Factory(
+        lambda self: OpenSearch(
+            hosts=[{'host': self.host, 'port': self.port}],
+            http_auth=self.http_auth,
+            use_ssl=self.use_ssl,
+            verify_certs=self.verify_certs,
+            connection_class=RequestsHttpConnection
+        ),
+        takes_self=True
+    ))
+
+    def upsert_vector(
+            self,
+            vector: list[float],
+            vector_id: Optional[str] = None,
+            namespace: Optional[str] = None,
+            meta: Optional[dict] = None,
+            **kwargs
+    ) -> str:
+
+        vector_id = vector_id if vector_id else utils.str_to_hash(str(vector))
+        doc = {
+            "vector": vector,
+            "namespace": namespace,
+            "metadata": meta
+        }
+        doc.update(kwargs)
+        response = self.client.index(index=self.index_name, id=vector_id, body=doc,
+                                     routing=namespace if namespace else "default")
+
+        return response["_id"]
+
+    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> Optional[BaseVectorStoreDriver.Entry]:
+        try:
+            response = self.client.get(index=self.index_name, id=vector_id, routing=namespace if namespace else None)
```

Comment:
Should we use `search` here as well and just do a `multi_match` on `id` and `namespace`?